### PR TITLE
code changes for pagination

### PR DESCRIPTION
**Pull request by reubenson**
Addresses https://github.com/slategroup/slate-web/issues/3731, as a follow-up to https://github.com/slategroup/slate-web/issues/2852.

With this PR, we follow Coral's guidance on implementing pagination on comment streams such that visitors to pages with very long comment threads (4000+) are not penalized by very slow render times. Following their suggestion for a solution to this was incomplete, however, due to the incompatibility between enabling `DATA_CACHE` (for Redis caching of comments) and pagination (which relies on querying Mongo). After some deliberation over our options, we arrived at the present PR, in which pagination is limited to comment streams that are over 500 comments long OR is archived. For all other cases, the Redis caching strategy enabled through the `DATA_CACHE` feature flag will be the pathway used to serve comments to readers, in which they will see all comments on page load.

The behavior described above diverges from the official release of Coral version 9.5.2 in that the official version will load all comments associated with the story on load. Our Open Thread pages commonly contain 20,000 comments, translating to a >20MB payload on page render. 

## Expected Behavior
- When a user visits an archived comment thread ([example](https://slate.com/briefing/2021/10/slates-open-thread-pepin-the-short.html)) with any number of comments, the comments will be paginated (they will see a `Load More` button)
- When a user visits a non-archived comment thread ([example](https://slate.com/comments/advice/2024/12/motherhood-freaky-part-care-and-feeding.html)) with 500 or fewer comments, all comments will load on page load.
- When a user visits a non-archived comment thread ([example](https://slate.com/briefing/2024/11/join-slates-open-thread-for-the-week-of-november-11.html)) with more than 500 comments, comments will be paginated.

### CODE REVIEW

File: client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabContainer.tsx [hunk 1]
```diff
@@ -244,27 +244,23 @@ export const AllCommentsTabContainer: FunctionComponent<Props> = ({
...
   const beginViewNewCommentsEvent = useViewerNetworkEvent(
     ViewNewCommentsNetworkEvent
   );
 
-  useEffect(() => {
-    setLocal({ commentsFullyLoaded: !hasMore });
-    if (hasMore && !isLoadingMore) {
-      void loadMoreAndEmit();
-    }
-  }, []);
```
Comment (reubenson) [inline client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabContainer.tsx:258]: removing this useEffect, which was responsible for querying for additional comments on page load

File: client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx [hunk 2]
```diff
@@ -164,35 +166,6 @@ const AllCommentsTabVirtualizedComments: FunctionComponent<Props> = ({
...
-                styles.loadAll
-              )}
-              // Added for keyboard shortcut support.
-              data-key-stop
-              data-is-load-more
-            >
-              Load All Comments
-            </Button>
-          </Localized>
-        )}
```
Comment (reubenson) [inline client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx:195]: Previously there were two different buttons that could load, depending on whether the user selected chronological or reverse-chron ordering of comments. Now simplified to a single Load More button

File: client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx [hunk 3]
```diff
@@ -222,11 +195,7 @@ const AllCommentsTabVirtualizedComments: FunctionComponent<Props> = ({
           top: increaseViewportBy,
           bottom: increaseViewportBy,
         }}
-        totalCount={
-          displayLoadAllButton
-            ? NUM_INITIAL_COMMENTS + newCommentsLength
-            : comments.length
-        }
+        totalCount={comments.length} // always render new comments as they get returned from GraphQL
```
Comment (reubenson) [inline client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx:198]: this is a special property of Virtuoso, which is responsible for lazily adding comments to the DOM based on visibility

File: server/src/core/server/config.ts [hunk 4]
```diff
@@ -505,6 +505,12 @@ const config = convict({
     default: ms("3000s"),
     env: "NOTIFICATIONS_POLL_RATE",
   },
+  pagination_count_threshold: {
+    doc: "the threshold number of comments above which pagination will be enabled",
+    format: Number,
+    default: 500,
```
Comment (reubenson) [inline server/src/core/server/config.ts:511]: This value has been experimentally derived, and seems reasonable for our performance constraints. If a page has 500 comments exactly, they will be fetched all at once with a GraphQL payload of ~500kb.

If we increase the threshold value above 500, users will increasingly be penalized with larger payloads on initial render.

If we decrease the threshold below 500, we will increase the number of queries that hit Mongo, requiring us to run the cluster with higher resources.

In general, this strategy of having a pagination threshold allows us to access the benefits of Redis caching without penalizing commenters on large comment pages, while also allowing us to run Coral on a lower resource tier for Mongo

File: server/src/core/server/graph/schema/schema.graphql [hunk 5]
```diff
@@ -4328,7 +4328,7 @@ Comment is a comment left by a [User](/api/types/objects/User) on an
 [Story](/api/types/objects/Story) or another [Comment](/api/types/objects/Comment)
 as a reply.
 """
-type Comment @cacheControl(maxAge: 5) {
+type Comment @cacheControl(maxAge: 30) {
```
Comment (reubenson) [inline server/src/core/server/graph/schema/schema.graphql:4331]: this increases the TTL for caching GraphQL queries, which will help system performance in the use case of very large, very active comment threads. this TTL is only applied when GRAPHQL_CACHE_HEADERS and GRAPHQL_RESPONSE_CACHE env vars are TRUE

File: client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx [hunk 6]
```diff
@@ -1,5 +1,4 @@
 import { Localized } from "@fluent/react/compat";
-import cn from "classnames";
```
Comment (elgreg) [inline client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx:2]: Just unused?

File: client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx [hunk 7]
```diff
@@ -21,8 +20,6 @@ import { AllCommentsTabVirtualizedCommentsLocal } from "coral-stream/__generated
 
 import AllCommentsTabCommentContainer from "./AllCommentsTabCommentContainer";
 
-import styles from "./AllCommentsTabVirtualizedComments.css";
```
Comment (elgreg) [inline client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx:24]: Just unused?

**Review by elgreg — COMMENTED** [review]
Not at a lot to review! 🎉

Is there a wiki/info on how to run this locally? I don't think I need to for this, but would be interesting to know how to do that so we can quickly test forks.

File: client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx (hunk 6 above)
Comment (reubenson) [reply client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx:2]: yup, no longer needed after simplifying the rendered template a bit

File: client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx (hunk 7 above)
Comment (reubenson) [reply client/src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/AllCommentsTabVirtualizedComments.tsx:24]: no longer needed!

**Comment by reubenson** [comment]
> Not at a lot to review! 🎉
> 
> Is there a wiki/info on how to run this locally? I don't think I need to for this, but would be interesting to know how to do that so we can quickly test forks.

I'm still in the process of updating the wiki here https://github.com/slategroup/slate-web/wiki/Comments-with-Coral-Talk. Alternatively I was thinking of updating the readme in the root directory of the repo, but it seems we generally leave those sparse in favor of wiki documentation?

**Comment by elgreg** [comment]
I don't really mind either way, but maybe go with the Wiki in this case since it will reduce conflicts when syncing the fork.

**Pull request merged**